tests/test_submit.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import pathlib
+from typing import TYPE_CHECKING
+
+import pytest
+from pydantic import BaseModel
+
+if TYPE_CHECKING:
+    from pytest_mock import MockerFixture
+
+class CheckoutGitTestScenario(BaseModel):
+    debug_scenario_name: str
+    git_status_mocked_output: str = ""
+    internet_permissions: list[str] = []
+    git_push_result: tuple[int, str] | None = None # (exit_code, output_message)
+    expected_prompts: list[str] = []
+
+@pytest.fixture(autouse=True)
+def clear_modules():
+    import sys
+    if "src.submit" in sys.modules:
+        del sys.modules["src.submit"]
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "scenario",
+    [
+        CheckoutGitTestScenario(
+            debug_scenario_name="no_changes_with_internet",
```

Comment:
Use `pytest.param(..., id="no_changes_with_internet")` for this instead, then you can also use it to select specific tests to run from the CLI (`pytest ... -t no_changes`)